File: Cadastro_de_alunos/app/src/main/java/com/example/brendon/registrodealunosv1/Adapters/FaculdadeAdapter.java
package com.example.brendon.registrodealunosv1.Adapters;

import android.content.Intent;
import android.support.design.widget.Snackbar;
import android.support.v7.widget.RecyclerView;
import android.support.annotation.NonNull;
import android.annotation.SuppressLint;
import android.content.DialogInterface;
import android.view.LayoutInflater;
import android.widget.ImageButton;
import android.app.AlertDialog;
import android.content.Context;
import android.widget.PopupMenu;
import android.widget.TextView;
import android.view.ViewGroup;
import android.view.View;

import com.example.brendon.registrodealunosv1.Formularios.FormularioAddAluno;
import com.example.brendon.registrodealunosv1.Formularios.FormularioAddFaculdade;
import com.example.brendon.registrodealunosv1.Models.Faculdade;
import com.example.brendon.registrodealunosv1.R;
import io.objectbox.Box;


public class FaculdadeAdapter extends RecyclerView.Adapter<FaculdadeAdapter.FaculdadeViewHolder> {

    public static String ID = "idFaculdade";

    private Context context;
    private Box<Faculdade> listFaculdades;

    public FaculdadeAdapter(Context context, Box<Faculdade> listFaculdades) {
        this.listFaculdades = listFaculdades;
        this.context = context;
    }

    @NonNull
    @Override
    public FaculdadeViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
        LayoutInflater inflater = LayoutInflater.from(context);
        View view = inflater.inflate(R.layout.card_faculdade, viewGroup, false);
        return new FaculdadeViewHolder(view);
    }

    @SuppressLint({"SetTextI18n","RecyclerView"})
    @Override
    public void onBindViewHolder(@NonNull FaculdadeViewHolder holder, final int position) {
        final Faculdade faculdadeAtual = this.listFaculdades.getAll().get(position);

        holder.txtNome.setText("Nome: " + faculdadeAtual.getNome());
        holder.txtEmail.setText("Email: " + faculdadeAtual.getEmail());
        holder.txtContato.setText("Contato " + faculdadeAtual.getContatoPrincipal());

        imageMenus(holder.itemView, faculdadeAtual, position);
        popMenu(holder.itemView, faculdadeAtual, position);
    }

    private void imageMenus(View view, Faculdade faculdade, int position){

        ImageButton excluir_faculdade = view.findViewById(R.id.excluir_faculdade);
        ImageButton editar_faculdade = view.findViewById(R.id.editar_faculdade);

        editar_faculdade.setOnClickListener(l ->{
            Intent intent = new Intent(context, FormularioAddFaculdade.class);
            intent.putExtra(ID, faculdade.id);
            context.startActivity(intent);
            notifyItemChanged(position);
        });

        excluir_faculdade.setOnClickListener(l ->{
            AlertDialog alerta;
            AlertDialog.Builder builder = new AlertDialog.Builder(context);
            builder.setTitle("Remover");
            builder.setMessage("Deseja remover : " + faculdade.getNome()+" ?");
            builder.setPositiveButton("Sim", (arg0, arg1) -> {
                listFaculdades.remove(faculdade);
                notifyItemRemoved(position);
                Snackbar.make(view,"Removido com sucesso!", Snackbar.LENGTH_LONG).show();
            });
            builder.setNegativeButton("Nao", (arg0, arg1) -> {
            });
            alerta = builder.create();
            alerta.show();
        });
    }

    private void popMenu(View view, Faculdade faculdade, int position){
        view.setOnLongClickListener((l ) -> {
            PopupMenu popup = new PopupMenu(context, view);
            popup.getMenuInflater().inflate(R.menu.pop_menu, popup.getMenu());

            popup.setOnMenuItemClickListener((item) -> {

                switch (item.getItemId()){
                    case R.id.op_editar:

                        Intent intent = new Intent(context, FormularioAddFaculdade.class);
                        intent.putExtra(ID, faculdade.id);
                        context.startActivity(intent);
                        notifyItemChanged(position);
                        break;

                    case R.id.op_remover:

                        AlertDialog alerta;
                        AlertDialog.Builder builder = new AlertDialog.Builder(context);
                        builder.setTitle("Remover");
                        builder.setMessage("Deseja remover a faculdade: " + faculdade.getNome()+" ?");
                        builder.setPositiveButton("Sim", (arg0, arg1) -> {
                            listFaculdades.remove(faculdade);
                            notifyItemRemoved(position);
                            Snackbar.make(l,"Removido com sucesso!", Snackbar.LENGTH_LONG).show();
                        });
                        builder.setNegativeButton("Nao", (arg0, arg1) -> {
                        });
                        alerta = builder.create();
                        alerta.show();
                        break;
                }

                return false;
            });

            popup.show();

            return true;
        });
    }

    @Override
    public int getItemCount() {
        return this.listFaculdades.getAll().size();
    }

    public class FaculdadeViewHolder extends RecyclerView.ViewHolder {
        TextView txtNome, txtEmail, txtContato;

        public FaculdadeViewHolder(View view) {
            super(view);
            txtNome = view.findViewById(R.id.view_nome_faculdade);
            txtEmail = view.findViewById(R.id.view_email);
            txtContato = view.findViewById(R.id.view_contato);

        }
}
}

File: RegistrodeAlunos v2/app/src/main/java/com/example/artur/registrodealunos/Adapter/FaculdadeAdapter.kt
package com.example.artur.registrodealunos.Adapter

import android.content.Context
import android.content.Intent
import android.support.design.widget.Snackbar
import android.support.v7.app.AlertDialog
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.PopupMenu
import android.widget.Toast
import com.example.artur.registrodealunos.FormularioFaculdadeActivity
import com.example.artur.registrodealunos.Modelo.Aluno
import com.example.artur.registrodealunos.Modelo.Faculdade
import com.example.artur.registrodealunos.R
import io.objectbox.Box
import kotlinx.android.synthetic.main.item_faculdade.view.*

class FaculdadeAdapter(private val context: Context,
                       private val faculdadeBox: Box<Faculdade>): RecyclerView.Adapter<FaculdadeAdapter.ViewHolder>(){

    companion object {
        const val ID = "idFaculdade"
    }

    class ViewHolder(itemView: View): RecyclerView.ViewHolder(itemView){
        fun bind(faculdade: Faculdade){
            val nome = itemView.text_nome_faculdade
            val email= itemView.text_email_faculdade

            nome.text = faculdade.nome
            email.text = faculdade.email
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, p1: Int): ViewHolder {
        return ViewHolder(LayoutInflater.from(context).inflate(R.layout.item_faculdade, parent, false))
    }

    override fun getItemCount(): Int {
        return faculdadeBox.all.size
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val faculdade = faculdadeBox.all[position]
        holder.bind(faculdade)

        imageMenus(holder.itemView, faculdade, position)
        popMenu(holder.itemView, faculdade, position)
    }

    private fun popMenu(itemView: View, faculdade: Faculdade, position: Int){
        itemView.setOnLongClickListener { view ->
            val popup = PopupMenu(context, view)
            popup.menuInflater.inflate(com.example.artur.registrodealunos.R.menu.pop_menu, popup.menu)
            popup.setOnMenuItemClickListener { item ->
                when (item.itemId){
                    R.id.op_editar -> popEditar(faculdade, position)
                    R.id.op_remover -> popExcluir(faculdade, position)

                }
                false
            }
            popup.show()

            true
        }
    }

    private fun popEditar(faculdade: Faculdade, position: Int){
        val intent = Intent(context, FormularioFaculdadeActivity::class.java)
        intent.putExtra(ID, faculdade.id)
        context.startActivity(intent)
        notifyItemChanged(position)
    }

    private fun popExcluir(faculdade: Faculdade, position: Int){
        val alertDialog = AlertDialog.Builder(context)

        alertDialog.setTitle("Excluir")
            .setMessage("Deseja realmente excluir ${faculdade.nome}?")
            .setPositiveButton("SIM"){_, _ ->
                this.faculdadeBox.remove(faculdade)
                notifyItemRemoved(position) //atualiza a lista
                notifyItemRangeChanged(position, itemCount)
                Toast.makeText(context, "${faculdade.nome} removida", Toast.LENGTH_LONG).show()
            }
            .setNegativeButton("NÃO"){_, _ ->}
            .create()
            .show()
    }

    private fun imageMenus(itemView: View, faculdade: Faculdade, position: Int){
        val editar = itemView.icon_editar_faculdade
        val excluir = itemView.icon_excluir_faculdade

        editar.setOnClickListener { viewEditar(faculdade, position) }

        excluir.setOnClickListener { viewExcluir(faculdade, position) }
    }

    private fun viewEditar(faculdade: Faculdade, position: Int){
        val intent = Intent(context, FormularioFaculdadeActivity::class.java)
        intent.putExtra(ID, faculdade.id)
        context.startActivity(intent)
        notifyItemChanged(position)
    }

    private fun viewExcluir(faculdade: Faculdade, position: Int){
        val alertDialog = AlertDialog.Builder(context)

        alertDialog.setTitle("Excluir")
            .setMessage("Deseja realmente excluir ${faculdade.nome}?")
            .setPositiveButton("SIM"){_, _ ->
                this.faculdadeBox.remove(faculdade)
                notifyItemRemoved(position) //atualiza a lista
                notifyItemRangeChanged(position, itemCount)
                Toast.makeText(context, "${faculdade.nome} removida", Toast.LENGTH_LONG).show()
            }
            .setNegativeButton("NÃO"){_, _ ->}
            .create()
            .show()
    }

}